diff --git a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,2 +1,2 @@
 [package]
-name = "mwpf"
+name = "mwpf_rational"
@@ -23,4 +23,4 @@
 [lib]
-name = "mwpf"
+name = "mwpf_rational"
 path = "src/lib.rs"
diff --git a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -5,2 +5,2 @@
 [project]
-name = "mwpf"
+name = "mwpf_rational"
@@ -31,5 +31,5 @@
 [tool.maturin]
 no-default-features = true
-features = ["python_binding", "f64_weight", "embed_visualizer"]
+features = ["python_binding", "rational_weight", "embed_visualizer"]
 bindings = "pyo3"
-cargo-extra-args = "--no-default-features --features python_binding,f64_weight,embed_visualizer"
+cargo-extra-args = "--no-default-features --features python_binding,rational_weight,embed_visualizer"
diff --git a/src/lib.rs b/src/lib.rs
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -64,3 +64,3 @@
 #[cfg(feature = "python_binding")]
 #[pymodule]
-fn mwpf(py: Python<'_>, m: &PyModule) -> PyResult<()> {
+fn mwpf_rational(py: Python<'_>, m: &PyModule) -> PyResult<()> {
